pycbc/inference/models/hierarchical.py

```diff
@@ -113,17 +115,18 @@ def __init__(self, variable_params, submodels, **kwargs):
             self.__extra_stats += self.extra_stats_map[lbl]
             # also make sure the model's sampling transforms and waveform
             # transforms are not set, as these are handled by the hierarchical
-            # model
-            if model.sampling_transforms is not None:
-                raise ValueError("Model {} has sampling transforms set; "
-                                 "in a hierarchical analysis, these are "
-                                 "handled by the hiearchical model"
-                                 .format(lbl))
-            if model.waveform_transforms is not None:
-                raise ValueError("Model {} has waveform transforms set; "
-                                 "in a hierarchical analysis, these are "
-                                 "handled by the hiearchical model"
-                                 .format(lbl))
+            # model, except for `joint_primary_marginalized` model
+            if self.name != "joint_primary_marginalized":
```

Comment:
Why is this here? Ideally anything different only exists in each subclass. Parent classes shouldn't have know about subclasses really.

Reply:
@ahnitz In the standard `hierarchical` model, all the transforms will be handled on the top level, but in our case, remember that we need to allow the transform with the prefix to be handled by submodels. Currently, I don't have a better idea. We can fix this in later patches.

Reply:
@ahnitz In the standard `hierarchical` model, all the transforms will be handled on the top level, but in our case, remember that we need to allow the transform with the prefix to be handled by submodels. Currently, I don't have a better idea. We can fix this in later patches.

Reply:
@ahnitz In the standard `hierarchical` model, all the transforms will be handled on the top level, but in our case, remember that we need to allow the transform with the prefix to be handled by submodels. Currently, I don't have a better idea. We can fix this in later patches.